integrations/tensorflow/e2e/math/math_test.py

```diff
@@ -0,0 +1,561 @@
+import collections
+import os
+from typing import Any, Dict, Sequence, Type, Union
+
+from absl import app
+from absl import flags
+import numpy as np
+from pyiree.tf.support import tf_test_utils
+from pyiree.tf.support import tf_utils
+import tensorflow.compat.v2 as tf
+
+FLAGS = flags.FLAGS
+
+# Controls the input signature, call kwargs, input generator and args for a
+# unittest of a tf.math function.
+Config = collections.namedtuple("Config",
+                                ["signature", "kwargs", "generator", "args"])
+# Use old default API for compatibility with Python 3.6.
+Config.__new__.__defaults__ = (None, dict(), None, None)
+
+
+def create_signature_from_args(config: Config) -> Config:
+  """Uses a Config's args to generate an signature for tf.function."""
+  if config.signature is not None:
+    raise ValueError(
+        "Tried to add a signature to a config that already has one.")
+  if config.args is None:
+    raise ValueError(
+        "Tried to generate a signature from a config without any args.")
+
+  signature = tf_utils.apply_function(
+      config.args, lambda x: tf.TensorSpec.from_tensor(tf.convert_to_tensor(x)))
+  return Config(signature, config.kwargs, config.generator, config.args)
+
+
+def configure_dtypes(base_config: Config,
+                     dtypes: Sequence[tf.DType],
+                     exported_name: str = None) -> Dict[str, Config]:
```

Comment:
This argument never appears to be used? It seems odd when coupled with the functionality that automatically prepends the function name to the dict keys when normalizing.

Reply:
The use case is was for something like this:

```python
"func":
    join_dicts(
        configure_dtypes(Config(settings_1), dtypes, "settings_1"),
        configure_dtypes(Config(settings_2), dtypes, "settings_2"),
    ),
```

But since `tf.math` is fairly simple there wasn't ever a need to test multiple kwarg configurations. The new `Config` class supports this via:

```python
"func": {
    "settings_1": Config(settings_1,... dtypes),
    "settings_2": Config(settings_2,... dtypes),
},
```